lib/common/schema_validator.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Common
+  class SchemaValidator
+    def initialize(response, schema)
+      @response = response
+      @schema = schema
+    end
+
+    def validate
+      errors = JSON::Validator.fully_validate(parsed_schema, parsed_response)
+
+      log_schema_errors(errors) if errors.any?
+    end
+
+    private
+
+    def parsed_response
+      JSON.parse(@response)
+    rescue JSON::ParserError => e
+      Rails.logger.error('Schema validator received invalid JSON response ', response: @response, details: e)
```

Comment:
For right now, data is only available through logging. I think another alternative could be to set up a slack channel and have it alert us by slack.